tests/test_homogeneous_collections.py

```diff
@@ -121,10 +121,97 @@ class RootModel(BaseXmlModel, tag='model'):
     assert_xml_equal(actual_xml, xml)
 
 
+def test_text_list_extraction():
+    class RootModel(BaseXmlModel, tag="model"):
+        values: List[int]
+
+    xml = '''
+    <model>1 2 70 -34</model>
+    '''
+
+    actual_obj = RootModel.from_xml(xml)
+    expected_obj = RootModel(
+        values=[1, 2, 70, -34]
+    )
+
+    assert actual_obj == expected_obj
+
+    actual_xml = actual_obj.to_xml()
+    assert_xml_equal(actual_xml, xml)
+
+
+def test_text_tuple_extraction():
+    class RootModel(BaseXmlModel, tag="model"):
+        values: Tuple[int, ...]
+
+    xml = '''
+    <model>1 2 70 -34</model>
+    '''
+
+    actual_obj = RootModel.from_xml(xml)
+    expected_obj = RootModel(
+        values=[1, 2, 70, -34]
+    )
+
+    actual_xml = actual_obj.to_xml()
+    assert_xml_equal(actual_xml, xml)
+
+
+def test_attr_list_extraction():
+    class RootModel(BaseXmlModel, tag="model"):
+        values: List[float] = attr()
+
+    xml = '''
+    <model values="3.14 -1.0 300.0"/>
+    '''
+    # This will fail if scientific notation is used
+    # i.e. if 300 is replaced with 3e2 or 300, the deserializer
+    # will always use the standard notation with the added `.0`.
+    # While this behaviour fails the tests, it shouldn't
+    # matter in practice.
+
+    actual_obj = RootModel.from_xml(xml)
+    expected_obj = RootModel(
+        values=[3.14, -1.0, 3e2]
+    )
+
+    assert actual_obj == expected_obj
+
+    actual_xml = actual_obj.to_xml()
+    assert_xml_equal(actual_xml, xml)
+
+
+def test_attr_tuple_extraction():
```

Comment:
the test is the same as previous.

Reply:
I'm testing both `tuple` and `list` to make sure that they both work. But yes, they are essentially the same and if this test is covered by other tests I'm willing to remove it.